src/box/retention_delay.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2023 VK Company Limited. All Rights Reserved.
+ *
+ * The information and source code contained herein is the exclusive property
+ * of VK Company Limited and may not be disclosed, examined, or reproduced in
+ * whole or in part without explicit written authorization from the Company.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "trivia/config.h"
+
+#if defined(ENABLE_RETENTION_DELAY)
+#include "retention_delay_impl.h"
+#else /* !defined(ENABLE_RETENTION_DELAY) */
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+#include "vclock/vclock.h"
+#include "xlog.h"
+
+/**
+ * Allocate vclock structure. In EE additional memory is allocated,
+ * where expiration time is saved. Note, that the returned value
+ * must be processed, as xmalloc is not used.
+ */
+static inline struct vclock*
+retention_vclock_create(void)
```

Comment:
Nit: this should be called `retention_vclock_new()`.

See https://www.tarantool.io/en/doc/latest/dev_guide/c_style_guide/#chapter-4-naming
```
For function naming we have a convention is to use:

    new/delete for functions which allocate + initialize and destroy + deallocate an object,
    create/destroy for functions which initialize/destroy an object but do not handle memory management,
    init/free for functions which initialize/destroy libraries and subsystems.

```

Reply:
Sure, fixed